### PR TITLE
WIP chore(core): moved special entity attributes into metadata

### LINKED ISSUE #2675: Add name and description fields to entities table (Trac #2675)
_Original ticket http://trac.elgg.org/ticket/2675 on 40894996-11-01 by ewinslow, assigned to unknown._

Elgg version: 1.7
- sites: name, description
- groups: name, description
- objects: title, description
- users: name, description (metadata)

Seems like theres a clear pattern.  objects' title could just as easily be called "name."

### LINKED ISSUE #8950: Get rid of entity->isFullyLoaded() and the tables_* properties
These should no longer be necessary as we use AttributeLoader.

### PR DESCRIPTION
**Pull request by jdalsem**
BREAKING CHANGE:
This removes the special entity tables groups_entity, users_entity and
objects_entity and puts the attributes into metadata. You can no longer
JOIN or QUERY these tables as they do not exist anymore. The attributes
have been moved into metadata so you can use
elgg_get_entities_from_metadata to query the attributes. You should not
be using elgg_get_entities_from_attributes anymore.

Reboot of #10629 

Refs #5459
Fixes #2675

- [ ] fix livesearch
- [ ] fix search
- [ ] write upgrade docs
- [ ] update datamodel docs
- [ ] check preloader features (after changing objects the query count increased)
- [ ] update class instances that get/set from private settings to use metadata for secondary attributes
- [x] get rid of `elgg_get_entities_from_attributes()` (or wire through ege_from_metadata)
- [x] fix user admin related functions (elgg_is_admin_user + ElggUser->*_admin)
- [ ] fix quick return in elgg_signed_request_gatekeeper
- [ ] what to do with user->last_action vs entity->last_action #5459 
- [x] fix #8950
- [ ] fix unittests
- [ ] benchmark
- [ ] test fresh install
- [ ] test, test, test, test

### CODE REVIEW

**Comment by jdalsem** [comment]
PR currently not functional

**Comment by hypeJunction** [comment]
Master is a bit broken I think. Something is off with rewrite rules.

**Comment by mrclay** [comment]
Reminder: This makes #4268 more dangerous, again suggesting ditching MD events. I like the idea of write-on-entity-save.

File: engine/lib/upgrades/2017072100-3.0.0-drop_entity_tables-3fa81afa7b402da9.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Elgg 3.0.0 upgrade 2016112500
+ * drop_entity_tables
+ *
+ * Migrates entity table data to metadata and drops the entity specific tables
```
Comment (mrclay) [inline engine/lib/upgrades/2017072100-3.0.0-drop_entity_tables-3fa81afa7b402da9.php]: Can upgrade.php actually boot to run this?

**Comment by mrclay** [comment]
Installer is writing to sites_entity.

**Comment by hypeJunction** [comment]
> I like the idea of write-on-entity-save.

I think it makes sense. It will be a nightmare to upgrade, but the benefits of having a full stack of changes will be benefitial.

**Comment by jdalsem** [comment]
This has been resolved via other PR's

**Pull request closed without merging**